packages/origin-247-certificate/src/offchain-certificate/repositories/CertificateReadModel/CertificateReadModelPostgres.repository.ts

```diff
@@ -15,7 +15,18 @@ export class CertificateReadModelPostgresRepository implements CertificateReadMo
     async save(
         certificateReadModel: CertificateReadModelEntity
     ): Promise<CertificateReadModelEntity> {
-        return await this.repository.save(certificateReadModel);
+        const existing = await this.getByInternalCertificateId(
+            certificateReadModel.internalCertificateId
+        );
+        if (existing) {
+            await this.repository.update(
```

Comment:
technically - without transaction this won't work entirely proper. 

But afair TypeORM `save` either updates or creates depending if item exists. Can you check documentation?

Reply:
Changed internalCertificateId to be primary column and used save instead.